### PR TITLE
Feature/cr 2809 crossdb ownership

### PR DESCRIPTION
**Pull request by idan-aviv**

### CODE REVIEW

**Review by Imanuel-Mizrahi — APPROVED** [review]

File: cloudrail/knowledge/rules/gcp/non_context_aware/sql_cross_databases_ownership_chaining_rule.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    def execute(self, env_context: GcpEnvironmentContext, parameters: Dict[ParameterType, any]) -> List[Issue]:
+        issues: List[Issue] = []
+        for sql_db in env_context.sql_database_instances:
+            if self.is_version_contains(sql_db, 'SQLSERVER') and \
+                    self.is_flag_on(sql_db, 'cross db ownership chaining'):
+                issues.append(
+                    Issue(
+                        f"The Google Cloud database instance `{sql_db.get_friendly_name()}` has database"
+                        f" flag ‘cross db ownership chaining’ set to on.",
```
Comment (Tomerva) [inline cloudrail/knowledge/rules/gcp/non_context_aware/sql_cross_databases_ownership_chaining_rule.py:22]: first time im seeing these 'front ticks'.. are those on purpose? (surrounding cross db ownership chaining)

**Review by Tomerva — APPROVED** [review]

File: cloudrail/knowledge/rules/gcp/non_context_aware/sql_cross_databases_ownership_chaining_rule.py (hunk 1 above)
Comment (idan-aviv) [reply cloudrail/knowledge/rules/gcp/non_context_aware/sql_cross_databases_ownership_chaining_rule.py:22]: @Imanuel-Mizrahi is it valid ticks? i just copied it as is...

Comment (Imanuel-Mizrahi) [reply cloudrail/knowledge/rules/gcp/non_context_aware/sql_cross_databases_ownership_chaining_rule.py:22]: That's a good catch @Tomerva 
Based on what I got this morning from Alon, this is not a valid tick.
But the code which should parse it is in the UI, so maybe there is more which I am not aware of...
Unfortunately, we do not have the script which checks how the CLI output will look like, so we cannot test the evidence.

File: cloudrail/knowledge/rules/gcp/gcp_rules_metadata.yaml [hunk 2]
```diff
@@ -156,4 +156,32 @@ rules_metadata:
...
+      Enabling cross db ownership is not recommended unless all of the databases hosted by the instance of SQL Server
+      must participate in cross- database ownership chaining.
+      This recommendation is applicable to SQL Server database instances. 
+    human_readable_logic: Cloudrail will review the database instance configuration within your GCP subscription and
+      Terraform plan to ensure that SQL Server database instances have ‘cross db ownership chaining’
+      database flag set to off.
+    remediation_steps:
+      terraform: For the google_sql_database_instance resources, ensure that ‘cross db ownership chaining’
+        database flag is set to off for SQL Server databases.
+      console: Follow the guide at https://cloud.google.com/sql/docs/sqlserver/flags in order to ensure that
```
Comment (doronafota) [inline cloudrail/knowledge/rules/gcp/gcp_rules_metadata.yaml:180]: missing <> around the url

File: cloudrail/knowledge/rules/gcp/non_context_aware/sql_cross_databases_ownership_chaining_rule.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        return 'non_car_cloud_sql_crossdb_ownership_chaining_on'
+
+    def execute(self, env_context: GcpEnvironmentContext, parameters: Dict[ParameterType, any]) -> List[Issue]:
+        issues: List[Issue] = []
+        for sql_db in env_context.sql_database_instances:
+            if self.is_version_contains(sql_db, 'SQLSERVER') and \
+                    self.is_flag_on(sql_db, 'cross db ownership chaining'):
+                issues.append(
+                    Issue(
+                        f"The Google Cloud database instance `{sql_db.get_friendly_name()}` has database"
```
Comment (doronafota) [inline cloudrail/knowledge/rules/gcp/non_context_aware/sql_cross_databases_ownership_chaining_rule.py:21]: replace database instance with get type

**Review by doronafota — APPROVED** [review]

**Pull request merged**